src/mpi/datatype/typerep/src/typerep_yaksa_pack.c

```diff
@@ -86,18 +86,20 @@ static int typerep_do_copy(void *outbuf, const void *inbuf, MPI_Aint num_bytes,
 
         yaksa_info_t info = MPII_yaksa_get_info(&inattr, &outattr);
         if (typerep_req) {
+            typerep_req->info = info;
```

Comment:
Was there actually an instance was it freed too early? The main thing I'm seeing here is that the info free for nonblocking typerep requests is moved into the test/wait functions. The caller is still responsible for freeing the info for blocking ops. Not sure the inconsistency is good.

Reply:
Disregard the comment on the info free. I think this change makes sense. You just need to make sure not to free info in the nonblocking case until completion. It might make sense that info gets copied by yaksa so we don't need to do this in the future. That's basically how MPI_Info objects work.

I'd still prefer if the MPIR_Typerep_test support was added in a separate commit. If you have some time to split that out, I think it would be better.

Reply:
Yes, the test function is now in a separate PR